README.adoc

```diff
@@ -78,29 +78,79 @@ checkout([$class: 'GitSCM',
           userRemoteConfigs: [[url: 'https://github.com/jenkinsci/git-client-plugin.git']]])
 ---
 
-==== Checkout with ssh and a private key credential
+==== Checkout with a shallow clone to reduce data traffic
 
-Insert example here
+Checkout from the git client plugin source repository using https no credentials, a default branch, and with a shallow clone which  requests a limited number of commits from the tip of the requested branch(es) thereby saving time, data transfer and disk space.
 
-==== Checkout with https and a username/password credential
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: '*/master']],
+          extensions: [[$class: 'CloneOption', noTags: false, reference: '', shallow: true]],
+          userRemoteConfigs: [[url: 'https://github.com/jenkinsci/ws-cleanup-plugin']]])
+---
 
-Insert example here
+==== Checkout with a narrow refspec
 
-==== Checkout with git large file support enabled
+Checkout from the git client plugin source repository using https no credentials, a default branch, and with refspec defined for the repository. This can save time, data transfer and disk space when you only need to access the references specified by the refspec.
```

Comment:
Since the repository URL is for ws-cleanup, need to make the text match the example.  This repository prefers sentence per line formatting in the documentation, so inserted a line break at the end of the sentence.
```suggestion
Checkout from the workspace cleanup plugin source repository using https no credentials, a default branch, and with refspec defined for the repository.
This can save time, data transfer and disk space when you only need to access the references specified by the refspec.
```